File: DragonIsland-Android/src/com/citex/dragonisland/core/tileset/Tile.java
package com.citex.dragonisland.core.tileset;

/**
 * This class is a level map tile description.
 * @author Lawrence Schmid
 */
public class Tile {

	/** Tile description index. */
	public int index; 
	
	/** Tile index. */
	public int tileIndex; 
	
	/** Tileset index. */
	public int tilesetIndex;   
	
	/** Tile number for tiled terain. */
	public int repeatIndex;  
	
	/** Tile state (0-normal, 1-collide, 2-empty). */
	public int state; 
	
	/** Item count */
	public int count; 
	
	/** Collision type. (0-none, 1-normal, 2-invisible) */
	public int collision; 
	
	/** Tile offset when hit. */
	public int offset; 
	
	/** Should the tile be drawn. */
	public int draw; 
	   
	/** Sprite tile description index. */
	public int spriteIndex;
	
	/** Sprite tile index. */
	public int spriteTile;
	
	/** Should the sprite tile be drawn. */
	public int spriteDraw;
	
	/** Tiled width. */
	public int width;
	
	/** Tiled height. */
	public int height;
	
	/** Direction of a sprite tile. */
	public char direction;
	
	/** Angle of a sprite tile. */
	public int angle;
	
	/**
	 * Gets whether tile is a gap.
	 * @return True if it is a gap; otherwise false.
	 */
	public boolean isGap() {
		
		if(collision == 0 || collision == 2 || collision == 3) 
			return true;
		else
			return false;
	
	}

}

File: DragonIsland-GP2X/Tile.cpp
#include "Tile.h"

/**
 * Default constructor
 */
Tile::Tile()
{
	index = 0;
	tile1 = 0;
	tile2 = 0;
	tile3 = 0;
	state = 0;
	count = 0;
	collision = 0;
	offset = 0;
	draw = 0;
	spriteIndex = 0;
	spriteTile = 0;
	spriteDraw = 0;

}

/**
 * Deallocates memory by destroying the Tile
 */
Tile::~Tile()
{
	delete &index;     
	delete &tile1;     
	delete &tile2;    
	delete &tile3;    
	delete &state;     
	delete &count;     
	delete &collision; 
	delete &offset;   
	delete &draw;      
	delete &spriteIndex;
	delete &spriteTile;
	delete &spriteDraw;
}
